test/feature/middleware_support.feature

```diff
@@ -0,0 +1,153 @@
+Feature: Middleware support
+
+  Scenario Outline: simple middleware support
+    Given a file "routes.coffee" with the content
+      """
+      module.exports = ({ GET }) ->
+        GET '/users', to: 'users#index'
+      """
+    And a file "controllers/users_controller.coffee" with the content
+      """
+      middle = (req, res, next) ->
+        req.custom = 'foo'
+        next()
+
+      class UsersController
+        middleware:
+          index: middle
+
+        index:   (req, res) -> res.end req.custom
```

Comment:
The main use case is to support 3rd party libraries like [Passport](http://passportjs.org/ "Passport").
The library expects you to add per-endpoint middleware. And I called it middleware because it is middleware.
Though it might be semantically nicer to call the helper in `BaseController` something like `beforeAction` instead of `useMiddleware` so it is similar to Rails `before_action`.